src/AbstractSerializer.php

```diff
@@ -137,11 +138,12 @@ protected static function serializeHeaders(array $headers): string
     /**
      * Filter a header name to wordcase
      *
-     * @param string $header
+     * @param string|int $header
      */
     protected static function filterHeader($header): string
     {
-        $filtered = str_replace('-', ' ', $header);
+        $filtered = is_int($header) ? (string) $header : $header;
```

Comment:
would just cast always to string, since we have to check `is_int`, directly casting int and string might be even "faster".